src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets

```
@@ -33,6 +33,12 @@ This file contains the .NET 5-specific targets to customize ILLink
         <Type>Microsoft.Android.Sdk.ILLink.SetupStep</Type>
       </_TrimmerCustomSteps>
     </ItemGroup>
+    <ItemGroup>
+      <_PreserveLists Include="$(MSBuildThisFileDirectory)..\PreserveLists\*.xml" />
+      <TrimmerRootDescriptor
+          Condition=" '@(ResolvedFileToPublish)' != '' and '%(Filename)' != '' "
```

I wouldn't do `'@(ResolvedFileToPublish)' != ''` as this can expand to a large string.

Maybe this works instead?
```suggestion
          Condition=" '@(ResolvedFileToPublish->Count())' != '0' and '%(Filename)' != '' "
```

---

Unfortunately that doesn't work. I think it breaks the way the msbuild iterates over the two item groups and so we don't get the intersection of them. When run with above change, the resulting item group contains all the preserve files:

```
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\Java.Interop.xml
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\Mono.Android.xml
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\mscorlib.xml
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\OpenTK.xml
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\System.Data.Services.Client.xml
                   Added Item(s): TrimmerRootDescriptor=C:\Users\rodo\git\xa-master\packages\microsoft.android.sdk\11.0.100-ci.master.145\targets\..\PreserveLists\System.ServiceModel.xml
17:31:17.746   1:8>Done building target "_PrepareLinking" in project "UnnamedProject.csproj".: (TargetId:251)
```

---

What about?
```suggestion
          Condition=" '%(ResolvedFileToPublish.Filename)' != '' "
```

---

Wait, I had the parentheses wrong :-) It works nicely, thanks for the improvement.